gc/base/standard/Scavenger.cpp

```diff
@@ -4435,6 +4435,11 @@ void
 MM_Scavenger::reportGCCycleFinalIncrementEnding(MM_EnvironmentStandard *env)
 {
 	OMRPORT_ACCESS_FROM_OMRPORT(env->getPortLibrary());
+	uintptr_t cycleType = env->_cycleState->_type;
+	/* set high bit(OMR_GC_CYCLE_TYPE_STATE_UNSUCCESSFUL) in the cycleType of CycleEnd event in scavenge backout case */
+	if ((OMR_GC_CYCLE_TYPE_SCAVENGE == cycleType) && env->getExtensions()->isScavengerBackOutFlagRaised()) {
```

Comment:
Can `cycleType` be different than `OMR_GC_CYCLE_TYPE_SCAVENGE` at this point? If no, we can replace check to assertion.

Reply:
I would not bother check the type and even asserting it's scavenge. whatever the type is, we are just adding info that it failed.

Reply:
could be even inlined in the trigger hook line